packages/runtime/container-runtime/src/blobManager.ts

```diff
@@ -391,13 +410,24 @@ export class BlobManager extends TypedEventEmitter<IBlobManagerEvents> {
 		);
 	}
 
-	private getBlobHandle(id: string): IFluidHandle<ArrayBufferLike> {
+	private getBlobHandle(id: string): BlobHandle {
 		assert(
 			this.redirectTable.has(id) || this.pendingBlobs.has(id),
 			0x384 /* requesting handle for unknown blob */,
 		);
-		return new BlobHandle(`${BlobManager.basePath}/${id}`, this.routeContext, async () =>
-			this.getBlob(id),
+		const pending = this.pendingBlobs.get(id);
+		const callback = () => {
+			// in detached container case, there is no pending entry
+			if (pending) {
```

Comment:
you shouldn't need to check pending in the callback. maybe move the below condition here
```
const callback = pending ? () => ... : undefined
```